src/numerics/BandMatrix.cpp

```diff
@@ -6,6 +6,7 @@
 #include "cantera/numerics/BandMatrix.h"
 #include "cantera/base/utilities.h"
 #include "cantera/base/stringUtils.h"
+#include "sundials/sundials_config.h"
```

Comment:
My general philosophy is not to rely on transitive imports like that, in case they get removed elsewhere. Is it more typical in C++ to rely on the transitive includes?

Reply:
I feel like trying to locally `#include` everything directly just leads to having a massive preamble in every source file, and there's no good way (AFAIK) to tell whether you've actually succeeded at avoiding transitive imports.

And at least in the case of a `config.h` file, I think it's a pretty standard expectation that that will be included directly or indirectly in every other header file for the project. On the other end of the spectrum, though, I wouldn't want to rely on a SUNDIALS header including a specific system header or header from another library.

Reply:
I think that makes sense. My point about avoiding transitive dependencies was only that I like to include all the direct dependencies for a file. In this case, since we're directly using the values from the `sundials_config.h`, I prefer to include (import in Python, which is where I mostly do this) that file. `sundials_config.h` can manage its own dependencies, if it has any. 

Point taken about config.h files in general though, I'll remove this!

Reply:
But if `CT_USE_LAPACK` is true, then the preprocessor directives that access `SUNDIALS_VERSION_MAJOR` are all in branches that are inactive, so it doesn't matter that it isn't defined.